src/ketu/async/source.clj

```diff
@@ -73,21 +73,22 @@
       (fn [consumer]
         (consumer/assign! consumer (consumer/topic-partitions topic partitions))))))
 
-(defn- poll-fn [^Consumer consumer opts]
-  (let [source-name (:ketu/name opts)
-        poll-timeout-duration (Duration/ofMillis (:ketu.source/poll-timeout-ms opts))
-        catching-poll? (:ketu.source.legacy/catching-poll? opts)]
-    (if catching-poll?
-      ;TODO Eliminate catching poll ASAP.
-      ; Just in case of a production issue and generic error handling wasn't implemented yet.
-      (fn []
-        (try
-          (consumer/poll! consumer poll-timeout-duration)
-          (catch Exception e
-            (log/error logger "[source={}] Caught poll exception" source-name e)
-            nil)))
-      (fn []
-        (consumer/poll! consumer poll-timeout-duration)))))
+(defn- poll-fn [^Consumer consumer should-poll? opts]
+  (when @should-poll?
+    (let [source-name           (:ketu/name opts)
+          poll-timeout-duration (Duration/ofMillis (:ketu.source/poll-timeout-ms opts))
+          catching-poll?        (:ketu.source.legacy/catching-poll? opts)]
+      (if catching-poll?
```

Comment:
With the suggested decorator, we'll be able to deprecate and remove the `:ketu.source.legacy/catching-poll?` in the near future.